src/components/Parsing.tsx

```diff
@@ -0,0 +1,383 @@
+  FilesContent: any,
+  DataModel: any,
+  ObjectPropertiesName: any
+) => {
+  // Pre-select the files for test purposes (to be removed once I'm done building/testing the function)
+  const FileUploaded = FilesUploaded.slice(0, 20); // 20 is the max length
+  const FileContent = FilesContent.slice(0, 20);
+
+  // Iterate on the list of files uploaded
+  // With the data uploader it will be easier to retrieve the content of the files/file names (properties of the object). For now, it relies on the fact that the objects in FilesUploaded and FilesContent have the same order.
+  for (let i = 0; i < FileContent.length; i++) {
+    // Define the array that will store the properties describing all the data points scanned
+    const aggArray = [];
+    // Check the type of file uploaded (.csv, .xlsx, .json, etc.)
+    // For now we focus only on JSON files uploaded from Facebook platform
+    if (FileUploaded[i].split('.')[1] === 'json') {
+      // Check if the file is empty
+      if (Object.keys(FileContent[i]).length === 0) {
+        console.log('empty');
+      } else {
+        // Get depth of the file scanned. The depth is defined manually (cf. DataModel) from the maximum number of steps it takes to get to the desired object
+        const fileDepth = (DataModel.datamodel as any)[FileUploaded[i]][
+          'file_structure_properties'
+        ]['depth'];
+
+        if (fileDepth === 0) {
+          const nestedArrayName = 'None'; // If the depth === 0 then necessarily, there is no nested array name
+
+          for (let j = 0; j < FileContent[i].length; j++) {
+            const indivArray = [];
+            for (let k = 0; k < ObjectPropertiesName.length; k++) {
+              indivArray.push(
+                (DataModel.datamodel as any)[FileUploaded[i]][nestedArrayName][
+                  'entries'
+                ][k][ObjectPropertiesName[k]]
+              );
+            }
+            aggArray.push(indivArray);
+          }
+        } else if (fileDepth === 1) {
+          const nestedArrayName = String(Object.keys(FileContent[i]));
+
+          if (typeof FileContent[i][nestedArrayName] === 'string') {
+            // Check the type of the element that comes right after the nested array name (string vs object). If it's a string, then it means that there is only one single data point to parse.
+            for (let j = 0; j < ObjectPropertiesName.length; j++) {
+              aggArray.push(
+                (DataModel.datamodel as any)[FileUploaded[i]][nestedArrayName][
+                  'entries'
+                ][j][ObjectPropertiesName[j]]
+              );
+            }
+          } else {
+            if (
+              (DataModel.datamodel as any)[FileUploaded[i]][
+                'file_structure_properties'
+              ]['nested_data_point_selector'] !== ''
+            ) {
+              // Check if the file has multiple nested array names and select the right one with nestedDataSelector property (cc. DataModel)
```

Comment:
Just to make sure you understand what the objective is to know how the json is composed ?
In the datamodel, we specify the path within the JSON that leads back to our values? So if the path exists we take the data if it exists it is that the data does not exist for the user? No ?

Reply:
I think I need a live demo to be sure to fully understand it